include/aspect/particle/property/interface.h

```diff
@@ -0,0 +1,355 @@
+          /**
+           * Read the parameters this class declares from the parameter file.
+           * The default implementation in this class does nothing, so that
+           * derived classes that do not need any parameters do not need to
+           * implement it.
+           */
+          virtual
+          void
+          parse_parameters (ParameterHandler &prm);
+      };
+
+
+
+      /**
+       * Manager class of properties - This class sets the data of the particles
+       * and updates it over time if requested by the user selected properties
+       */
+      template <int dim>
+      class Manager : public SimulatorAccess<dim>
+      {
+        public:
+          /**
+           * Empty constructor for Manager
+           */
+          Manager ();
+
+          /**
+           * Destructor for Manager
+           */
+          virtual
+          ~Manager ();
+
+          /**
+           * Initialization function. This function is called once at the
+           * beginning of the program after parse_parameters is run.
+           */
+          virtual
+          void
+          initialize ();
+
+          /**
+           * Initialization function for particle properties. This function is
+           * called once at the creation of a particle
+           */
+          virtual
+          void
+          initialize_particle (Particle<dim> &particle,
+                               const Vector<double> &solution,
+                               const std::vector<Tensor<1,dim> > &gradients);
+
+          /**
+           * Update function for particle properties. This function is
+           * called once every timestep for every particle
+           */
+          virtual
+          void
+          update_particle (Particle<dim> &particle,
+                           const Vector<double> &solution,
+                           const std::vector<Tensor<1,dim> > &gradients);
+
+          /**
+           * Returns a bool, which is false if no selected plugin needs to
+           * update tracer properties over time. This saves considerable
+           * computation time in cases, when no plugin needs to update tracer
+           * properties over time, because the solution does not need to be
+           * evaluated at tracer positions in this case.
+           */
+          virtual
+          UpdateTimeFlags
+          need_update ();
+
+          /**
+           * Get the number of doubles required to represent this particle's
+           * properties for communication.
+           *
+           * @return Number of doubles required to represent this particle
+           */
+          unsigned int
+          get_data_len () const;
+
+          /**
+           * Get the size in number of bytes required to represent this
+           * particle's properties for communication.
+           *
+           * @return Number of bytes required to represent this particle
+           */
+          std::size_t
+          get_particle_size () const;
```

Comment:
This function has (essentially) the same documentation as `get_data_len()`. I think their documentations should be adjusted to reflect what exactly it is they do, and how they differ.